include/boost/fusion/container/vector/vector.hpp

```diff
@@ -9,12 +9,620 @@
+                  std::size_t
+                , fusion::result_of::size<typename remove_reference<Sequence>::type>::value
+              >
+        {};
+
+        // Object proxy since preserve object order
+        template <std::size_t, typename T>
+        struct store
+        {
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store()
+                : elem()
+            {}
+
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store(store const& rhs)
+                : elem(rhs.get())
+            {}
+
+            //BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            //store(T const& val)
+            //    : elem(val)
+            //{}
+
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store&
+            operator=(store const& rhs)
+            {
+                elem = rhs.get();
+                return *this;
+            }
+
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store(store&& rhs)
+                : elem(std::forward<T>(rhs.get()))
+            {}
+
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store(typename remove_reference<T>::type&& val)
+                : elem(std::forward<T>(val))
+            {}
+
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store&
+            operator=(store&& rhs)
+            {
+                elem = std::forward<T>(rhs.get());
+                return *this;
+            }
+
+            template <typename U>
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store(U&& rhs
+                , typename enable_if<
+                      mpl::or_<
+                          is_convertible<typename traits::remove_rvalue_reference<U>::type, T>
+                        , detail::is_constructible<T, typename traits::remove_rvalue_reference<U>::type>
+                      >
+                  >::type* = 0)
+                : elem(std::forward<U>(rhs))
+            {}
+
+            template <typename U>
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            store&
+            operator=(U&& rhs)
+            {
+                elem = std::forward<U>(rhs);
+                return *this;
+            }
+
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            T      & get()       { return elem; }
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            T const& get() const { return elem; }
+
+            T elem;
+        };
+
+        template <std::size_t I, typename... T>
+        struct data;
+
+        template <std::size_t I>
+        struct data<I>
+        {
+            typedef mpl::int_<(int)I> size;
+
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            void
+            assign() BOOST_NOEXCEPT
+            {}
+        };
+
+        template <std::size_t I, typename T, typename... Tail>
+        struct data<I, T, Tail...>
+            : store<I, T>, data<I + 1, Tail...>
```

Comment:
> libc++ expands the tuple using multiple inheritance, and libstdc++ uses recursive inheritance (similar to this implementation). When using multiple inheritance, larger tuples don't hit the recursive template expansion limit, since the expansion occurs at the same depth level. 

OK, I'll take a look the implementation.